typeset/converter.js

```diff
@@ -239,7 +243,48 @@ async function injectCoverageCollection (page) {
   }
 }
 
+const highlightCodeElements = (inputPath) => {
+  let inputFile
+
+  try {
+    const data = fs.readFileSync(inputPath, 'utf8')
+    inputFile = data.toString()
+  } catch (err) {
+    console.error(err)
+  }
+
+  const dom = new JSDOM(inputFile, {
+    contentType: "application/xhtml+xml"
+  })
+
+  const preTagElements = [ ... dom.window.document.querySelectorAll("pre") ] 
+  preTagElements.forEach(pre => {
+    const highlightedCode = hljs.highlightAuto(pre.textContent).value  
```

Comment:
`highlightAuto` works now! But the snippet needs to be wrapped in the `code` tag with the class of the syntax highlighting you want to use: 

`<pre><code class="html">...</code></pre>`

I think next I will test nesting two `code` tags in one `pre` tag with different languages. For example, in the content we were working with yesterday, we removed the `html` because the `highlightAuto` didn't know what to do with all of the extra tags. 

Also as I'm writing this comment I'm also realizing I still have to add the minified highlight.js stylesheet in at some point lol

Reply:
I think it is important to not rely on`highlightAuto` to detect the correct language; instead, the language should be specified as an attribute in the `<pre>` tag.

I am confused what is meant by "nesting two code tags"... CNXML only allows one language (via the `lang="..."` attribute [defined here](https://legacy.cnx.org/eip-help/code))